eips_code/eip-3540.py

```diff
@@ -6,85 +6,92 @@
 S_DATA = 0x02
 
 
-def validate_eof(code: bytes):
-    # Old-style contracts are still allowed
-    if len(code) == 0 or code[0] != FORMAT:
-        return
+# Determines if code is in EOF format of any version.
+def is_eof(code: bytes) -> bool:
+    return len(code) >= 2 and code[0] == FORMAT and code[1] == MAGIC
+
 
-    # Validate format and magic
-    assert(len(code) >= 3 and code[1] == MAGIC and code[2] == VERSION)
+# Validate EOF code.
+def validate_eof(code: bytes):
+    # Check version
+    assert len(code) >= 3 and code[2] == VERSION
 
     # Process section headers
     section_sizes = {S_CODE: 0, S_DATA: 0}
     pos = 3
     while True:
         # Terminator not found
-        assert(pos < len(code))
+        assert pos < len(code)
         section_id = code[pos]
         pos += 1
         if section_id == S_TERMINATOR:
             break
 
         # Disallow unknown sections
-        assert(section_id in section_sizes)
+        assert section_id in section_sizes
 
         # Data section preceding code section
-        assert(not (section_id == S_DATA and section_sizes[S_CODE] == 0))
+        assert section_id != S_DATA or section_sizes[S_CODE] != 0
 
         # Multiple sections with the same id
-        assert(section_sizes[section_id] == 0)
+        assert section_sizes[section_id] == 0
 
         # Truncated section size
-        assert((pos + 1) < len(code))
+        assert (pos + 1) < len(code)
         section_sizes[section_id] = (code[pos] << 8) | code[pos + 1]
         pos += 2
 
         # Empty section
-        assert(section_sizes[section_id] != 0)
+        assert section_sizes[section_id] != 0
 
     # Code section cannot be absent
-    assert(section_sizes[S_CODE] != 0)
+    assert section_sizes[S_CODE] != 0
 
     # The entire container must be scanned
-    assert(len(code) == (pos + section_sizes[S_CODE] + section_sizes[S_DATA]))
+    assert len(code) == (pos + section_sizes[S_CODE] + section_sizes[S_DATA])
+
 
+# Validates any code
 def validate_code(code: bytes) -> bool:
-    try:
-        validate_eof(code)
-        return True
-    except:
-       return False
+    if is_eof(code):
+        try:
+            validate_eof(code)
+        except:
+            return False
+    return True
+
 
 # Legacy contracts
 assert validate_code(b'') == True
 assert validate_code(b'\x00') == True
-# No magic
-assert validate_code(b'\xef') == False
+assert validate_code(b'\xef') == True  # Magic byte missing
+
```

Comment:
Remove this line, so that "Any value outside the magic" stays part of "Legacy contracts" section